src/properties.rs

```diff
@@ -42,176 +52,200 @@ pub const CFSTATS: &str = property!("CFSTATS");
+pub const AGGREGATED_TABLE_PROPERTIES: &CStr = property!("aggregated-table-properties");
 
 /// "rocksdb.aggregated-table-properties-at-level<N>", same as the previous
 /// one but only returns the aggregated table properties of the
 /// specified level "N" at the target column family.
-pub const AGGREGATED_TABLE_PROPERTIES_AT_LEVEL: &str =
-    property!("aggregated-table-properties-at-level");
+pub fn aggregated_table_properties_at_level(level: usize) -> CString {
+    unsafe { level_property("aggregated-table-properties-at-level", level) }
+}
 
 /// "rocksdb.actual-delayed-write-rate" - returns the current actual delayed
 /// write rate. 0 means no delay.
-pub const ACTUAL_DELAYED_WRITE_RATE: &str = property!("actual-delayed-write-rate");
+pub const ACTUAL_DELAYED_WRITE_RATE: &CStr = property!("actual-delayed-write-rate");
 
 /// "rocksdb.is-write-stopped" - Return 1 if write has been stopped.
-pub const IS_WRITE_STOPPED: &str = property!("is-write-stopped");
+pub const IS_WRITE_STOPPED: &CStr = property!("is-write-stopped");
 
 /// "rocksdb.estimate-oldest-key-time" - returns an estimation of
 /// oldest key timestamp in the DB. Currently only available for
 /// FIFO compaction with
 /// compaction_options_fifo.allow_compaction = false.
-pub const ESTIMATE_OLDEST_KEY_TIME: &str = property!("estimate-oldest-key-time");
+pub const ESTIMATE_OLDEST_KEY_TIME: &CStr = property!("estimate-oldest-key-time");
 
 /// "rocksdb.block-cache-capacity" - returns block cache capacity.
-pub const BLOCK_CACHE_CAPACITY: &str = property!("block-cache-capacity");
+pub const BLOCK_CACHE_CAPACITY: &CStr = property!("block-cache-capacity");
 
 /// "rocksdb.block-cache-usage" - returns the memory size for the entries
 /// residing in block cache.
-pub const BLOCK_CACHE_USAGE: &str = property!("block-cache-usage");
+pub const BLOCK_CACHE_USAGE: &CStr = property!("block-cache-usage");
 
 /// "rocksdb.block-cache-pinned-usage" - returns the memory size for the
 /// entries being pinned.
-pub const BLOCK_CACHE_PINNED_USAGE: &str = property!("block-cache-pinned-usage");
+pub const BLOCK_CACHE_PINNED_USAGE: &CStr = property!("block-cache-pinned-usage");
 
 /// "rocksdb.options-statistics" - returns multi-line string
 /// of options.statistics
-pub const OPTIONS_STATISTICS: &str = property!("options-statistics");
+pub const OPTIONS_STATISTICS: &CStr = property!("options-statistics");
+
+/// Constructs a property name for an ‘at level’ property.
+///
+/// `name` is the infix of the property name (e.g. `"num-files-at-level"`) and
+/// `level` is level to get statistics of.  The property name is constructed as
+/// `"rocksdb.<name><level>"`.
+///
+/// Expects `name` not to contain any interior NUL bytes.
+unsafe fn level_property(name: &str, level: usize) -> CString {
+    let bytes = format!("rocksdb.{}{}\0", name, level).into_bytes();
+    // SAFETY: We’re appending terminating NUL and we’re expecting user not to
+    // pass us a string with interior NUL bytes.
```

Comment:
I didn't get a comment about a user. This function is private and is only called from the library code, so this is indeed safe.

I think it would be better to place this (or a modified) comment into the function description as it is usually done in the standard library.

Reply:
User is anyone who calls the function.  In this case all the users are internal.  I’ve updated the comment to mention that.

The docstring of the function actually has that requirement as well.  It’s the last line.

The confusing thing here is that the `CString::from_vec_with_nul_unchecked` call should be wrapped inside of an unsafe block (and then this `SAFETY` comment would relate to that) but that currently results in ‘unnecessary `unsafe` block’ warning since [unsafe_block_in_unsafe_fn](https://rust-lang.github.io/rfcs/2585-unsafe-block-in-unsafe-fn.html) isn’t a thing yet.

Reply:
> The docstring of the function actually has that requirement as well. It’s the last line.

Yes, but in the standard library it is described in a separate section. For example here: https://doc.rust-lang.org/src/core/str/mod.rs.html#331
I think that safety invariants are more clear this way.

Anyway, I have approved your pull request, thanks for making this library better!